src/install.js

```diff
@@ -75,21 +109,47 @@ async function downloadSAPUI5(downloadURL, downloadPath) {
 
     response.data.pipe(fs.createWriteStream(zipFile))
     return new Promise((resolve, reject) => {
-      response.data.on('end', () => { resolve(zipFile) })
-      response.data.on('error', () => { reject() })
+      response.data.on('end', () => {
+        resolve(zipFile)
+      })
+      response.data.on('error', () => {
+        reject()
+      })
     })
   } catch (downloadError) {
     throw new Error(`Couldn't download SAPUI5 zip archive from '${downloadURL}'`)
   }
 }
 
 async function extractArchive(archive, targetPath) {
-  console.log('Extracting...')
-  const zip = new AdmZip(archive)
-  zip.extractAllTo(targetPath, true)
+  console.log(`Extracting files to  ${targetPath}`)
+
+  const zip = new StreamZip({
+    file: archive,
+    storeEntries: true,
+  })
+
+
+  zip.on('ready', () => {
+    console.log(`All entries read: ${zip.entriesCount}`)
+
+    const progressBar1 = new ProgressBar('Extracting: [:bar]', {
+
+      total: zip.entriesCount,
+    })
+    zip.on('extract', () => {
+      progressBar1.tick()
+      if (progressBar1.complete) {
+        console.log('SAPUI5 Installed')
+      }
+    })
+    zip.extract(null, targetPath, () => {
+      zip.close()
+    })
+  })
```

Comment:
As mentioned in `package.json` I'm not sure why you changed the module from AdmZip to StreamZip. Does it have a performance improvement?

Reply:
Please see my comments above